### PR TITLE
Data kind

### PR DESCRIPTION
**Pull request by jmbabazi**

### CODE REVIEW

File: configuration/pih/liquibase/sql/create_functions.sql [hunk 1]
```diff
@@ -2127,3 +2129,54 @@ RETURN ret;
...
+
+BEGIN
+    DECLARE creatorName TEXT;
+
+    SELECT CONCAT(given_name, ' ', family_name) creator_names INTO creatorName
+    FROM person_name pn
+    JOIN users u ON pn.voided = 0 AND u.retired = 0 AND pn.person_id = u.person_id
+    JOIN encounter e ON e.creator = u.user_id AND e.voided = 0 AND e.encounter_id = _encounter_id;
+
+    RETURN creatorName;
```
Comment (mogoodrich) [inline configuration/pih/liquibase/sql/create_functions.sql:2153]: Each user likely has only one name, but what happens if for some reason a user has two names?

File: configuration/pih/liquibase/sql/create_functions.sql [hunk 2]
```diff
@@ -2127,3 +2129,54 @@ RETURN ret;
...
+
+BEGIN
+    DECLARE creatorName TEXT;
+
+    SELECT DISTINCT(CONCAT(given_name, ' ', family_name)) creator_names INTO creatorName
+    FROM person_name pn
+    JOIN users u ON pn.voided = 0 AND u.retired = 0 AND pn.person_id = u.person_id
+    JOIN obs o ON o.creator = u.user_id AND o.voided = 0 AND o.encounter_id = _encounter_id;
+
+    RETURN creatorName;
```
Comment (mogoodrich) [inline configuration/pih/liquibase/sql/create_functions.sql:2179]: Same question here as on encounter creator

File: configuration/pih/petl/jobs/data_kind/encounter/source.sql [hunk 3]
```diff
@@ -0,0 +1,14 @@
+SELECT 
+    patient_id,
+    encounter_id,
+    encounter_type,
```
Comment (mogoodrich) [inline configuration/pih/petl/jobs/data_kind/encounter/source.sql:4]: encounter_type in the encounter table is just an int (foreign keying to the encounter_type table)... don't we want some method like ENCOUNTER_TYPE_NAME used here to resolve the actual name of the encounter_type?

File: configuration/pih/petl/jobs/data_kind/encounter/job.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+type: "sqlserver-bulk-import"
+configuration:
+  extract:
+    datasource: "mysql/openmrs.yml"
+    query:  "data_kind/encounter/source.sql"
+
+  load:
+    datasource: "sqlserver/openmrs_extractions.yml"
+    table: "datakind_encounter"
+    schema: "data_kind/encounter/target.sql"
```
Comment (mogoodrich) [inline configuration/pih/petl/jobs/data_kind/encounter/job.yml:10]: I'm wondering if we should think about adding a condition here to only run on JJ Dossen and Mirebalais?  

I created a ticket about adding support for this: https://pihemr.atlassian.net/browse/PETL-100

**Review by mogoodrich — COMMENTED** [review]
Generally looks good @jmbabazi , though I added a couple questions.

File: configuration/pih/liquibase/sql/create_functions.sql (hunk 1 above)
Comment (jmbabazi) [reply configuration/pih/liquibase/sql/create_functions.sql:2154]: thanks @mogoodrich , I added in a preffered = 1 condition

File: configuration/pih/petl/jobs/data_kind/encounter/source.sql (hunk 3 above)
Comment (jmbabazi) [reply configuration/pih/petl/jobs/data_kind/encounter/source.sql]: fixed

**Pull request merged**